File: src/main/java/com/example/Car.java
package com.example;

/**
 * Created by vctran on 08/03/16.
 */
public class Car {
    private Long id;

    private Brand brand;

    private Chassis chassis;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Brand getBrand() {
        return brand;
    }

    public void setBrand(Brand brand) {
        this.brand = brand;
    }

    public Chassis getChassis() {
        return chassis;
    }

    public void setChassis(Chassis chassis) {
        this.chassis = chassis;
    }
}

File: src/main/resources/static/js/car/car.js
/**
 * Created by vctran on 25/03/16.
 */
angular.module('car', []).controller('carController', function($scope, $http, Car, $rootScope, master, brands, $state) {
    $scope.master = master;
    $scope.brands = brands;
    $scope.car = new Car();
    $scope.car.user = $rootScope.user;
    $scope.addCar = function() {
        $scope.car.chassis = $scope.currentChassis;
        $scope.car.$save(function(){
            $scope.master = $scope.master.concat($scope.car);
            $scope.car = new Car();
        });
    };
    $scope.getChassis = function() {
        $http.get('getChassis', {params: {brandId :$scope.currentBrand.id}}).success(function(data){
            $scope.chassisList = data;
        });
    };
    $scope.set = function(car) {
        $state.go('main.carDetails', {car_id: car.id});
    }
}).controller('cardetailsController', function($scope, car, motors, escList, Setting, settings) {
    $scope.car = car;
    $scope.motors = motors;
    $scope.escList = escList;
    $scope.settings = settings;
    $scope.addSetup = function(car) {
        $scope.setting = new Setting();
        $scope.setting.motor = $scope.currentMotor;
        $scope.setting.esc = $scope.currentEsc;
        $scope.setting.car = car;
        $scope.setting.$save(function(){
            
        });
    };
});